plugins/tickets.py

```diff
@@ -0,0 +1,2118 @@
+
+                await self.user.send(msg)
+
+                # Publish the ticket
+                # Implicitly triggers update of the last ticket message
+                await self.current_ticket.publish()
+
+                # Schedule ticket expiration, if required
+                if duration is not None:
+                    update_expiry_for(self.current_ticket)
+
+                # Deliver the next ticket
+                await self.deliver()
+            else:
+                self.last_read_msgid = message.id
+
+
+async def reload_mods():
+    """
+    Reload all moderators from data.
+    """
+    global _ticketmods
+    logger.debug("Loading ticket moderators.")
+
+    # Unload mods
+    for mod in _ticketmods.values():
+        mod.unload()
+
+    # Rebuild ticketmod list
+    _ticketmods = {row[0]: TicketMod(row) for row in TicketMod._select_where()}
+
+    # Load mods
+    for mod in _ticketmods.values():
+        await mod.load()
+
+    logger.info("Loaded {} ticket moderators.".format(len(_ticketmods)))
+
+
+def get_or_create_mod(modid) -> TicketMod:
+    """
+    Get a single TicketMod by modid, or create it if it doesn't exist.
+    """
+    mod = _ticketmods.get(modid, None)
+    if not mod:
+        mod = TicketMod(TicketMod._insert(modid=modid))
+        _ticketmods[modid] = mod
+    return mod
+
+
+# ------------ Commands ------------
+
+def resolve_ticket(msg, args, rewind=False) -> Ticket:
+    """
+    Resolves a ticket from the given message and command args, if possible.
+    Ticket is extracted from either the referenced message or the first arg.
+    """
+    ticket = None
+    if ref := msg.reference:
+        if (ref_msg := ref.resolved) and isinstance(ref_msg, discord.Message):
+            if ref_msg.author == client.user and ref_msg.embeds:
+                embed = ref_msg.embeds[0]
+                if (name := embed.author.name) and name.startswith("Ticket #"):
+                    ticket_id = int(name[8:].split(' ', maxsplit=1)[0])
+                    ticket = get_ticket(ticket_id)
+    if ticket is None:
+        rewind_pos = args.pos
+        ticketarg = args.next_arg()
+        if ticketarg is not None and isinstance(ticketarg, commands.StringArg):
+            maybe_id = int(ticketarg.text)
+            if maybe_id < 2147483647:  # Hack to differentiate msgid from tid
+                tickets = fetch_tickets_where(id=maybe_id)
+            else:
+                tickets = fetch_tickets_where(list_msgid=maybe_id)
+            ticket = next(tickets, None)
+        if rewind and ticket is None:
+            args.pos = rewind_pos
+    return ticket
+
+
+def summarise_tickets(*tickets, title="Tickets", fmt=None):
+    """
+    Create paged embeds of ticket summaries from the provided list of tickets.
+    """
+    if not tickets:
+        return None
+
+    lines = [ticket.summary(fmt=fmt) for ticket in tickets]
+    blocks = ['\n'.join(lines[i:i+10]) for i in range(0, len(lines), 10)]
```

Comment:
Is 10 an empirically tested number? I figure the summary length doesn't vary much, but maybe consider trying to fit as many as possible into a page, given the embed size limits.

P.S. I've been thinking about some generalized functionality that solves the problem of fitting things of varying lengths under certain length limits.